src/python/gem5/resources/resource.py

```diff
@@ -554,6 +565,122 @@ def _get_weights_and_simpoints_from_file(
         return simpoint_list, weight_list
 
 
+class SuiteResource(AbstractResource):
+    """
+    A suite resource. This resource is used to specify a suite of workloads to
+    run on a board. It contains a list of workloads to run, along with their
+    IDs and versions.
+
+    Each workload in a suite is used to create a `WorkloadResource` object.
+    These objects are stored in a list and can be iterated over.
+    """
+
+    def __init__(
+        self,
+        workload_obj_list: List[Tuple["WorkloadResource", str]] = [],
```

Comment:
A better solution `workloads: Dict["WorkloadResource", Set[str]] = {}`.

Improvements:

1. Rename `workload_obj_list` to `workloads`.
2. Removes the confusing "List of tuples".
3. Allows for quick association of workload to group_label.